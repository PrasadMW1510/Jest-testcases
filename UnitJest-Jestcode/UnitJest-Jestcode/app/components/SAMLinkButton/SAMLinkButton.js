/**
 *
 * SamlinkButton
 *
 */

import PropTypes from 'prop-types';
import React from 'react';
import { withRouter } from 'react-router';

import SAMButton from 'components/SAMButton';

export function SAMLinkButton(props) {
  const { history, suppressRedirect, to } = props;

  const onClickHandler = () => {
    let onClickResult;
    if (props.onClickHandler) {
      // note, if the user didn't return anything from onClickHandler,
      // onClickResult will be 'undefined'.
      onClickResult = props.onClickHandler();
    }
    // onClickHandler has to explicitly return 'false' && suppressRedirect must be false if it wants to suppress redirect
    if (onClickResult !== false && !suppressRedirect) {
      history.push(to);
    }
  };

  return (
    <SAMButton
      id={props.id}
      onClickHandler={onClickHandler}
      disabled={props.disabled}
      isPrimaryButton={props.isPrimaryButton}
      buttonClassModifier={props.buttonClassModifier}
      buttonType={props.buttonType}
    >
      {props.children}
    </SAMButton>
  );
}

SAMLinkButton.defaultProps = {
  buttonType: 'button',
  disabled: false,
  isPrimaryButton: false,
  suppressRedirect: false,
};

SAMLinkButton.propTypes = {
  children: PropTypes.node.isRequired,
  to: PropTypes.string.isRequired,
  suppressRedirect: PropTypes.bool,
  buttonClassModifier: PropTypes.string,
  buttonType: PropTypes.string,
  disabled: PropTypes.bool,
  history: PropTypes.object,
  id: PropTypes.string,
  isPrimaryButton: PropTypes.bool,
  onClickHandler: PropTypes.func,
};

export default withRouter(SAMLinkButton);
