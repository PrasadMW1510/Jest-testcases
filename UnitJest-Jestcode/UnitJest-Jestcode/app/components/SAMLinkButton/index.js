/**
 *
 * SAMLinkButton
 *
 */

export { default } from './SAMLinkButton';
