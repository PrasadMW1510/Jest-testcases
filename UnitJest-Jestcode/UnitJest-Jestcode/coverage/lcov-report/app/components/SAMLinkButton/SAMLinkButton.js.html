<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/components/SAMLinkButton/SAMLinkButton.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">app/components/SAMLinkButton</a> SAMLinkButton.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/23</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/11</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">/**<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ></span></span>
 *
 * SamlinkButton
 *
 */
&nbsp;
import PropTypes from 'prop-types';
import React from 'react';
import { withRouter } from 'react-router';
&nbsp;
import SAMButton from 'components/SAMButton';
&nbsp;
export function SAMLinkButton(props) {<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span></span></span></span></span></span>
  const { history, suppressRedirect, to } = props;
&nbsp;
  const onClickHandler = () =&gt; {
    let onClickResult;<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span></span></span></span>
<span class="cstat-no" title="statement not covered" >    if (props.onClickHandler) {</span>
      // note, if the user didn't return anything from onClickHandler,
      // onClickResult will be 'undefined'.
<span class="cstat-no" title="statement not covered" >      onClickResult = props.onClickHandler();</span>
    }
    // onClickHandler has to explicitly return 'false' &amp;&amp; suppressRedirect must be false if it wants to suppress redirect
<span class="cstat-no" title="statement not covered" >    if (onClickResult !== false &amp;&amp; !suppressRedirect) {</span>
<span class="cstat-no" title="statement not covered" >      history.push(to);</span>
    }
  };
&nbsp;
<span class="cstat-no" title="statement not covered" >  return (</span>
    &lt;SAMButton
      id={props.id}
      onClickHandler={onClickHandler}
      disabled={props.disabled}
      isPrimaryButton={props.isPrimaryButton}
      buttonClassModifier={props.buttonClassModifier}
      buttonType={props.buttonType}
    &gt;
      {props.children}
    &lt;/SAMButton&gt;
  );
}
&nbsp;
<span class="cstat-no" title="statement not covered" >SAMLinkButton.defaultProps = {</span>
  buttonType: 'button',
  disabled: false,
  isPrimaryButton: false,
  suppressRedirect: false,
};
&nbsp;
<span class="cstat-no" title="statement not covered" >SAMLinkButton.propTypes = {</span>
  children: PropTypes.node.isRequired,
  to: PropTypes.string.isRequired,
  suppressRedirect: PropTypes.bool,
  buttonClassModifier: PropTypes.string,
  buttonType: PropTypes.string,
  disabled: PropTypes.bool,
  history: PropTypes.object,
  id: PropTypes.string,
  isPrimaryButton: PropTypes.bool,
  onClickHandler: PropTypes.func,<span class="cstat-no" title="statement not covered" ></span>
};
&nbsp;
export default withRouter(SAMLinkButton);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Jul 21 2018 17:06:47 GMT+0530 (India Standard Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
