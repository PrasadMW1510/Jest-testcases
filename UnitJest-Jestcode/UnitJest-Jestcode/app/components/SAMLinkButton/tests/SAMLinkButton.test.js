import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

import { SAMLinkButton } from '../SAMLinkButton';

describe('<SamlinkButton />', () => {
  let wrapper = null;
  let mockOnClickHandler = null;
  let mockHistory = null;
  const fakeEvent = { preventDefault: () => {} };

  beforeEach(() => {
    mockOnClickHandler = jest.fn();
    mockHistory = {
      push: jest.fn(),
    };
  });

  it('expect the default to render', () => {
    wrapper = shallow(<SAMLinkButton to="/mockLocation1">mockButton1</SAMLinkButton>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('expect primary button to render', () => {
    wrapper = shallow(
      <SAMLinkButton to="/mockLocation2" isPrimaryButton>
        mockButton2
      </SAMLinkButton>
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('onClickHandler returning undefined (default)', () => {
    wrapper = shallow(
      <SAMLinkButton to="/mockLocation3" history={mockHistory} onClickHandler={mockOnClickHandler}>
        mockButton3
      </SAMLinkButton>
    );
    wrapper.find('SAMButton').prop('onClickHandler')(fakeEvent);
    expect(mockOnClickHandler).toHaveBeenCalled();
    expect(mockHistory.push).toHaveBeenCalledWith('/mockLocation3');
  });

  it('onClickHandler returning false', () => {
    const mockOnClickHandlerReturningFalse = jest.fn().mockReturnValue(false);
    wrapper = shallow(
      <SAMLinkButton
        to="/mockLocation3"
        history={mockHistory}
        onClickHandler={mockOnClickHandlerReturningFalse}
      >
        mockButton3
      </SAMLinkButton>
    );
    wrapper.find('SAMButton').prop('onClickHandler')(fakeEvent);
    expect(mockOnClickHandlerReturningFalse).toHaveBeenCalled();
    expect(mockHistory.push).toHaveBeenCalledTimes(0);
  });

  it('onClickHandler is not passed as props', () => {
    wrapper = shallow(
      <SAMLinkButton to="/mockLocation4" history={mockHistory}>
        mockButton4
      </SAMLinkButton>
    );

    wrapper.find('SAMButton').prop('onClickHandler')(fakeEvent);
    expect(mockOnClickHandler).not.toHaveBeenCalled();
    expect(mockHistory.push).toHaveBeenCalledWith('/mockLocation4');
  });
});
