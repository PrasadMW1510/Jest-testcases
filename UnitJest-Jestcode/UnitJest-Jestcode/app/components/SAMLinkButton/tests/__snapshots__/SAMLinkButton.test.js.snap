// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SamlinkButton /> expect primary button to render 1`] = `
<SAMButton
  buttonType="button"
  disabled={false}
  isPrimaryButton={true}
  onClickHandler={[Function]}
>
  mockButton2
</SAMButton>
`;

exports[`<SamlinkButton /> expect the default to render 1`] = `
<SAMButton
  buttonType="button"
  disabled={false}
  isPrimaryButton={false}
  onClickHandler={[Function]}
>
  mockButton1
</SAMButton>
`;
